client/src/templates/Challenges/utils/frame.ts

```diff
@@ -273,11 +273,47 @@ const initTestFrame = (frameReady?: () => void) => (frameContext: Context) => {
 const initMainFrame =
   (_: unknown, proxyLogger?: ProxyLogger) => (frameContext: Context) => {
     waitForFrame(frameContext)
-      .then(() => {
+      .then(async () => {
+        const iframeBuild: string = frameContext.build;
+        const parser = new DOMParser();
+        const iframeBuildDoc = parser.parseFromString(iframeBuild, 'text/html');
+        const allLinks = iframeBuildDoc.querySelectorAll(
+          'script[src], link[href]'
+        );
+        let errMsgs = '';
+        let unfetchedLinks = allLinks.length;
+
+        if (!unfetchedLinks) return null;
+
+        for (const url of allLinks) {
+          const urlAddr =
+            url.tagName === 'SCRIPT'
+              ? (url as HTMLScriptElement).src
+              : (url as HTMLLinkElement).href;
+
+          try {
+            const response = await fetch(String(urlAddr));
+            if (!response.ok) {
+              errMsgs = `${errMsgs}\nCannot retrieve ${urlAddr}, link status code: ${response.status}`;
+            } else if (!unfetchedLinks) {
+              continue;
+            }
+            unfetchedLinks -= 1;
+          } catch {
+            errMsgs = `${errMsgs}\n${urlAddr} does not exist. Only files that can be sourced are styles.css, script.js, or remote files`;
+          }
+        }
+
+        return errMsgs.length ? errMsgs : 'This link does not exist.';
+      })
+      .then(importResult => {
+        if (importResult) {
+          console.warn.call(frameContext.window, importResult);
+          proxyLogger && proxyLogger(importResult);
```

Comment:
```suggestion
          // @ts-expect-error the window types will be updated in another PR
          frameContext.window.console.error(importResult);
```
Unfortunately the types are wrong, but I'm already working on a PR that fixes them, so no need to worry about that here!

Anyways, we can make use of the fact that the frame's console already handles the proxying.

Reply:
Thanks for letting me know.